### PR TITLE
Add new /getsubstring command

### PR DESCRIPTION
**Pull request by chickemeeple**

### CODE REVIEW

**Comment by chickemeeple** [comment]
Essentially just copied /chat, and /variable as they are very similar. /chat and /variable and a few others only work because of the difference of string.split() and string.slice()

File: packages/junon-io/server/commands/getsubstring.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const BaseCommand = require("./base_command")
+
+class GetSubstring extends BaseCommand {
+    getUsage() {
+        return [
+            "Returns a variable named $substring",
+            "/getsubstring [substring] [message]",
```
Comment (simpleyuji) [inline packages/junon-io/server/commands/getsubstring.js]: something like `/getnthword [index] [message]`     (index is the correct programmer term for numbers that reference position of something)

File: packages/junon-io/server/commands/getsubstring.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            "Returns a variable named $substring",
+            "/getsubstring [substring] [message]",
+            "Ex: /getsubstring 2 a random message, creates a $substring variable, and assigns random to it."
+        ]
+    }
+    allowOwnerOnly() {
+        return true;
+    }
+    perform(caller,args) {
+        let substringSelector = args[0];
```
Comment (simpleyuji) [inline packages/junon-io/server/commands/getsubstring.js]: the variable should be called index

File: packages/junon-io/server/commands/getsubstring.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        if(isNaN(substringSelector)) {
+            caller.showChatError(`Substring selector cannot be NaN`)
```
Comment (simpleyuji) [inline packages/junon-io/server/commands/getsubstring.js]: same here `index must be a number`

File: packages/junon-io/server/commands/getsubstring.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            return
+        }
+        let text = args.slice(1).join(" ")
+        
+        let substrings = text.split(" ")
+
+        if(substringSelector > substrings.length+1) {
+            caller.showChatError(`Invalid substring selector.`);
```
Comment (simpleyuji) [inline packages/junon-io/server/commands/getsubstring.js]: something like `invalid index`

**Comment by chickemeeple** [comment]
Thank you for the corrections. I'll edit them so they are easier for people to understand.

**Review by simpleyuji — APPROVED** [review]

**Pull request merged**